app/main.py

```diff
@@ -1,8 +1,17 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result_list = [Person(person["name"], person["age"]) for person in people]
+    for index, person in enumerate(people):
+        if "wife" in person.keys() and person["wife"] is not None:
```

Comment:
will be better to use `get`:
```suggestion
       if person.get("wife") is not None:
```